### PR TITLE
[ECO-2587] Add market metadata on market page

### PR DESCRIPTION
**Pull request by CRBl69**
# Test

Go to the vercel preview then to `/dev/metadata`.

Set some metadata for your market.

Go to your market and check that everything is all right.

Alternatively, check eagle money.

### CODE REVIEW

**Comment by CRBl69** [comment]
Change contract to coin

**Review by xbtmatt — CHANGES_REQUESTED** [review]
1. Remove set copied, since it was a holdover from the emoji copy paste design
2. Try to remove the market cap/volume numbers from re-scrambling when clicking "Copy address'

Then this looks good to me! Looks great

File: src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx [hunk 1]
```diff
@@ -52,6 +82,61 @@ const MainInfo = ({ data }: MainInfoProps) => {
...
   const [copied, setCopied] = useState(false);
 
+  const borderStyle = "border-solid border-[1px] border-dark-gray rounded-[3px] p-[1em]";
+
+  const [marketProperties, setMarketProperties] = useState<Map<string, string>>();
+
+  const { aptos } = useAptos();
+
+  useEffect(() => {
+    MarketProperties.submit({
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx:92]: You can change this function name to `view` btw, meant to change it when generating the APIs

File: src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx [hunk 2]
```diff
@@ -141,10 +204,30 @@ const MainInfo = ({ data }: MainInfoProps) => {
...
+                if (!copied) {
+                  setCopied(true);
+                  setTimeout(() => setCopied(false), 3000);
+                }
+              }}
+              whileTap={{ scaleX: 0.96, scaleY: 0.98 }}
+              transition={{ ease: "linear", duration: 0.05 }}
+            >
+              <Button scale="lg" isScramble={false}>
+                copy contract address
```
Comment (alnoki) [inline src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx:222]:
```suggestion
                copy coin address
```

File: src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx (hunk 1 above)
Comment (CRBl69) [reply src/typescript/frontend/src/components/pages/emojicoin/components/main-info/MainInfo.tsx]: Addressed in 6ebc11c1a46c8e30cae4600360b55137e38a6a26.

File: src/typescript/frontend/src/utils/emoji.tsx [hunk 3]
```diff
@@ -10,26 +10,36 @@ declare global {
...
+        set?: string;
       };
     }
   }
 }
 
 export const Emoji = ({
   emojis,
+  set = undefined,
+  size = undefined,
```
Comment (xbtmatt) [inline src/typescript/frontend/src/utils/emoji.tsx:22]:
```suggestion
  size = "1em",
```

File: src/typescript/frontend/src/utils/emoji.tsx [hunk 4]
```diff
@@ -10,26 +10,36 @@ declare global {
...
+        key={`${emojisInString[i]}-${i}`}
+        size={size ?? "1em"}
+        native={e}
+        set={set}
+      ></em-emoji>
     ));
   } else {
     data = emojis.map((e, i) => (
-      <em-emoji key={`${emojis[i].emoji}-${i}`} size="1em" native={e.emoji}></em-emoji>
+      <em-emoji key={`${emojis[i].emoji}-${i}`} size={size ?? "1em"} native={e.emoji}></em-emoji>
```
Comment (xbtmatt) [inline src/typescript/frontend/src/utils/emoji.tsx:42]:
```suggestion
      <em-emoji key={`${emojis[i].emoji}-${i}`} size={size} native={e.emoji}></em-emoji>
```

File: src/typescript/frontend/src/utils/emoji.tsx [hunk 5]
```diff
@@ -10,26 +10,36 @@ declare global {
...
+  size?: string;
 }) => {
   let data: React.ReactNode[] = [];
   if (typeof emojis === "string") {
     const emojisInString = getEmojisInString(emojis);
     data = emojisInString.map((e, i) => (
-      <em-emoji key={`${emojisInString[i]}-${i}`} size="1em" native={e}></em-emoji>
+      <em-emoji
+        key={`${emojisInString[i]}-${i}`}
+        size={size ?? "1em"}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/utils/emoji.tsx:35]:
```suggestion
        size={size}
```

**Review by xbtmatt — APPROVED** [review]

**Pull request merged**